std/format/internal/write.d

```diff
@@ -925,6 +925,12 @@ if (is(FloatingPointTypeOf!T) && !is(T == enum) && !hasToString!(T, Char))
     assert(format("%a", r) == "0x1p-20");
 }
 
+// https://issues.dlang.org/show_bug.cgi?id=21836
+@safe pure unittest
```

Comment:
`@nogc`?

Reply:
Not `@nogc` since some code paths in the called function allocate, even though those paths are not taken when called with the argument in the unit test.

Reply:
do we have an `assertGCNotCalled` test helper?

Reply:
There is a `@nogc`-test in `std.format.internal.floats`. I think, it can be made `@nogc` once the three followups on this are merged and formatting floats is completely `@nogc`, not only zeros, nans and infs.